src/library_wasmfs.js

```diff
@@ -102,6 +102,14 @@ FS.createPreloadedFile = FS_createPreloadedFile;
     },
     // TODO: create
     // TODO: close
+    close: (fd) => {
+      var err = __wasmfs_close(fd);
+      if(err == {{{ cDefs.EBADF}}}) {
+        // throw new FS.ErrnoError({{{ cDefs.EBADF }}});
```

Comment:
I think it would be better to use `FS.ErrnoError` but it seems that `FS.ErrnoError` currently does not exist in `library_wasmfs.js`. I looked in `library_fs.js` and found `ErrnoError` here

https://github.com/emscripten-core/emscripten/blob/673b8464be8ec35472243e04ec3a525151d6fdc0/src/library_fs.js#L1414-L1460 

However, it seems like this method gets called in `init` or `staticInit`, both of which I do not see in `library_wasmfs.js`. Do you guys have any suggestions on adding `FS.ErrnoError` functionality to `library_wasmfs.js`?

Reply:
I think this is a case where we can replace the older complicated JS with something simpler. I mean that in `library_wasmfs` we can do
```js
function WasmFSError(code) {
  this.code = code;
}
```
and then do `throw new WasmFSError(code);`

In particular, it's ok to be simpler here because we don't need the `errno` handling. `errno` is already set by the C++ code that we call. So we just need a JS error to be thrown.